sdk/src/main/scala/io/horizen/storage/SidechainSecretStorage.scala

```diff
@@ -153,4 +153,25 @@ class SidechainSecretStorage(storage: Storage, sidechainSecretsCompanion: Sidech
     storage.update(new ByteArrayWrapper(Utils.nextVersion), updateList, removeList)
     this
   }
+
+  def storeSecretAndNonceAtomic(secret: SidechainTypes#SCS, nonce: Int, keyTypeSalt: Array[Byte]): Try[SidechainSecretStorage] = Try  {
+    require(nonce >= 0, "Nonce must be not negative")
+    require(keyTypeSalt != null, "Key type salt must be NOT NULL")
+    require(secret != null, "Can not add to storage: Secret must be NOT NULL.")
+
+    val key = calculateKey(secret.publicImage())
+    val value = new ByteArrayWrapper(sidechainSecretsCompanion.toBytes(secret))
+    require(!secrets.contains(key), "Key already exists - " + secret)
```

Comment:
This check can be moved before calculating value.

Reply:
Done